src/functional/swing_slide_filter.zig

```diff
@@ -0,0 +1,552 @@
+        Line{ .slope = -1, .intercept = 100 },
+    };
+
+    var list_values = std.ArrayList(f64).init(alloc);
+    defer list_values.deinit();
+    var compressed_values = ArrayList(u8).init(alloc);
+    defer compressed_values.deinit();
+    var decompressed_values = ArrayList(f64).init(alloc);
+    defer decompressed_values.deinit();
+    const error_bound: f32 = 0.0;
+
+    var i: usize = 0;
+    var lineIndex: usize = 0;
+    while (i < 100) : (i += 1) {
+        lineIndex = i / 25;
+        try list_values.append(f80Tof64(evaluate(&lines[lineIndex], i)));
+    }
+
+    const uncompressed_values = list_values.items;
+
+    try compress_swing(uncompressed_values[0..], &compressed_values, error_bound);
+    try decompress(compressed_values.items, &decompressed_values);
+
+    try testing.expect(mem.eql(f64, uncompressed_values[0..], decompressed_values.items));
+}
+
+test "swing-filter 0 error bound even size compress and decompress" {
+    const alloc = testing.allocator;
+    var line = Line{ .slope = 1, .intercept = 0.0 };
+
+    var list_values = std.ArrayList(f64).init(alloc);
+    defer list_values.deinit();
+    var compressed_values = ArrayList(u8).init(alloc);
+    defer compressed_values.deinit();
+    var decompressed_values = ArrayList(f64).init(alloc);
+    defer decompressed_values.deinit();
+    const error_bound: f32 = 0.0;
+
+    var rnd = std.rand.DefaultPrng.init(0);
+
+    var i: usize = 0;
+    while (i < 10) : (i += 1) {
+        const noise = rnd.random().float(f64) * 0.1 - 0.05;
+        try list_values.append(f80Tof64(evaluate(&line, i) + noise));
+    }
+
+    const uncompressed_values = list_values.items;
+
+    try compress_swing(uncompressed_values[0..], &compressed_values, error_bound);
+    try decompress(compressed_values.items, &decompressed_values);
+
+    try testing.expect(isWithinErrorBound(uncompressed_values, decompressed_values.items, error_bound));
+}
+
+test "swing-filter noisy line with 0 error bound and (odd size) compress and decompress" {
+    const alloc = testing.allocator;
+
+    var line = Line{ .slope = 1, .intercept = 0.0 };
+
+    var list_values = std.ArrayList(f64).init(alloc);
+    defer list_values.deinit();
+    var compressed_values = ArrayList(u8).init(alloc);
+    defer compressed_values.deinit();
+    var decompressed_values = ArrayList(f64).init(alloc);
+    defer decompressed_values.deinit();
+    const error_bound: f32 = 0.0;
+
+    var rnd = std.rand.DefaultPrng.init(0);
+
+    var i: usize = 0;
+    while (i < 11) : (i += 1) {
+        const noise = rnd.random().float(f64) * 0.1 - 0.05;
+        try list_values.append(f80Tof64(evaluate(&line, i) + noise));
+    }
+
+    const uncompressed_values = list_values.items;
+
+    try compress_swing(uncompressed_values[0..], &compressed_values, error_bound);
+    try decompress(compressed_values.items, &decompressed_values);
+
+    try testing.expect(isWithinErrorBound(uncompressed_values, decompressed_values.items, error_bound));
```

Comment:
Maybe place the arguments on separate line(s) as the line seems to [be above 100 characters](https://ziglang.org/documentation/master/#Whitespace).